buildSrc/src/main/java/org/elasticsearch/gradle/info/GlobalBuildInfoPlugin.java

```diff
@@ -213,26 +214,67 @@ private Object createGitRevisionResolver(final Project project) {
+                         * https://git-scm.com/docs/git-worktree.
+                         *
+                         * There are two cases to consider:
+                         *  - a plain repository with .git directory at the root of the working tree
+                         *  - a worktree with a plain text .git file at the root of the working tree
+                         *
+                         * In each case, our goal is to parse the HEAD file to get either a ref or a bare revision (in the case of being in
+                         * detached HEAD state).
+                         *
+                         * In the case of a plain repository, we can read the HEAD file directly, resolved directly from the .git
+                         * directory.
+                         *
+                         * In the case of a worktree, we read the gitdir from the plain text .git file. This resolves to a directory from
+                         * which we read the HEAD file and resolve commondir to the plain git repository.
+                         */
+                        final Path dotGit = project.getRootProject().getRootDir().toPath().resolve(".git");
+                        final Path head;
+                        final Path gitDir;
+                        if (Files.isDirectory(dotGit)) {
+                            // this is a git repository, we can read HEAD directly
+                            head = dotGit.resolve("HEAD");
+                            gitDir = dotGit;
+                        } else {
+                            // this is a git worktree, follow the pointer to the repository
+                            final Path workTree = Paths.get(readFirstLine(dotGit).substring("gitdir:".length()).trim());
+                            head = workTree.resolve("HEAD");
+                            final Path commonDir = Paths.get(readFirstLine(workTree.resolve("commondir")));
+                            if (commonDir.isAbsolute()) {
+                                gitDir = commonDir;
+                            } else {
+                                // this is the common case
+                                gitDir = workTree.resolve(commonDir);
+                            }
+                        }
+                        final String ref = readFirstLine(head);
+                        final String revision;
+                        if (ref.startsWith("ref:")) {
+                            revision = readFirstLine(gitDir.resolve(ref.substring("ref:".length()).trim()));
+                        } else {
+                            // we are in detached HEAD state
+                            revision = ref;
+                        }
+                        this.gitRevision.compareAndSet(null, revision);
+                    } catch (final IOException e) {
+                        // for now, do not be lenient until we have better understanding of real-world scenarios where this happens
```

Comment:
We have a number of Gradle integration tests that run in a workspace that is *not* an initialized Git repository. This causes [those tests to fail](https://gradle-enterprise.elastic.co/s/6qkkewlbiv77e/tests/orjfcz7hu5lqq-u46z5u74yfscg).

I think we should at least build in leniency that if this isn't a git repo (`.git` is missing) we just just carry on our merry way.

Reply:
Keep in mind that there are external users of the `BuildPlugin` that may have different development processes, or may distribute source distributions in the form of archives that would not contain VCS metadata.

Reply:
Yes, I tracked that down earlier from the failing CI builds here but I haven't pushed a fix for it yet.